src/platform/plugins/shared/dashboard/common/dashboard_container/types.ts

```diff
@@ -11,6 +11,17 @@ import type { Reference } from '@kbn/content-management-utils';
 
 import type { GridData } from '../../server/content_management';
 
+export interface DashboardSectionMap {
+  [id: string]: DashboardSectionState;
+}
+
+export interface DashboardSectionState {
+  title: string;
+  collapsed: boolean;
+  order: number;
+  id?: string;
```

Comment:
why is id optional? How is it possible to have a section without an id? How would panels be placed in this section?

Reply:
I was following the logic of `controls` where the user shouldn't have to provide an ID when creating a section via the dashboard API. cc @nickpeihl just in case I did this wrong 🙈

Reply:
ID is no longer optional on the front-end - see 69421ca7a7b5bbe768c32bb51db998d60e72be16